### PR TITLE
docs(documentation): improve process crash

### PR DESCRIPTION
**Pull request by hekike**
I incorporated the docs from the first deep dive (process crash) to our guides:
https://docs.google.com/document/d/1gd_4hlDK-DacbyTnFcqLPdCjM1Q1z02MdkFaUddPLCE/edit?ts=5de804b8#

### CODE REVIEW

File: documentation/abnormal_termination/README.md [hunk 1]
```diff
@@ -39,11 +43,49 @@ To find out if the process exited because of a `process.exit()` call, follow [Us
...
+Some Node.js application uses native modules to re-use existing components or
+performance optimize specific areas. A common example for native modules are
+database drivers. In the case our native module has an issue it can crash the
+Node.js application which will generate a native stack trace.
 
-### Process exits with unmeaningful Stack Trace
+You can find the most common scenarios for native crashes here:
 
-TBD
+1. Native module crashes due bug
```
Comment (Flarna) [inline documentation/abnormal_termination/README.md]: This is quite generic and includes point 2 and 3.

File: documentation/abnormal_termination/README.md [hunk 2]
```diff
@@ -39,11 +43,49 @@ To find out if the process exited because of a `process.exit()` call, follow [Us
...
+2. Native module crashes due memory leak (native memory leak)
+3. Bugs in Node.js runtime and dependencies
+  - For example: V8 checks, when input checking is missing can lead to crash
```
Comment (Flarna) [inline documentation/abnormal_termination/README.md]: checks in v8 are actually no bugs. missing validation in user code is not a bug in Node.js or it's dependencies.

File: documentation/abnormal_termination/README.md (hunk 1 above)
Comment (hekike) [reply documentation/abnormal_termination/README.md]: How about if the points would be:

1. Operational error in Node.js runtime and native dependencies
  - For example, V8 checks, when input checking is missing can lead to a crash
2. Programmer error in native code
  - For example, accessing incorrect memory address
3. Native module crashes due to memory leak (native memory leak)

File: documentation/abnormal_termination/README.md (hunk 2 above)
Comment (hekike) [reply documentation/abnormal_termination/README.md]: how about?

Operational error in Node.js runtime and native dependencies
  - For example, V8 checks, when input checking is missing can lead to a crash

Comment (Flarna) [reply documentation/abnormal_termination/README.md]: Maybe `For example, V8 checks, when input data checking to v8 APIs checking is missing can lead to a crash`.

File: documentation/abnormal_termination/README.md (hunk 1 above)
Comment (Flarna) [reply documentation/abnormal_termination/README.md]: Yes thats better. Other cases I have seen several times:
* Access garbage collected data
* Call v8 APIs without `HandleScope`
* Call v8 APIs from wrong Thread

Comment (hekike) [reply documentation/abnormal_termination/README.md]: Thank you, added

File: documentation/abnormal_termination/README.md (hunk 2 above)
Comment (hekike) [reply documentation/abnormal_termination/README.md]: Changed

File: documentation/abnormal_termination/README.md [hunk 3]
```diff
@@ -39,11 +43,53 @@ To find out if the process exited because of a `process.exit()` call, follow [Us
...
 
-TBD
+1. Native module crashes due bug
+2. Native module crashes due memory leak (native memory leak)
+3. Bugs in Node.js runtime and dependencies, for example:
+   - V8 checks, when input data checking to v8 APIs is missing can lead to a crash
+   - Access garbage collected data
+   - Call v8 APIs without HandleScope
+   - Call v8 APIs from wrong Thread
+   - Use native API compiled on linux using glibc on Alpine which uses Musl
```
Comment (gireeshpunathil) [inline documentation/abnormal_termination/README.md]: - accessing NULL pointer
 - attempting to process invalid file descriptors
?

Comment (hekike) [reply documentation/abnormal_termination/README.md]: Thanks, I added them

**Pull request merged**